cclib/parser/gaussianparser.py

```diff
@@ -1853,46 +1857,86 @@ def natural_orbital_single_spin_parsing(inputfile, updateprogress_title):
-           line[1:37] == "Mulliken charges and spin densities:" or \
-           line[1:32] == "Mulliken atomic spin densities:":
-
-            has_spin = 'spin densities' in line 
-            has_charges = 'charges' in line 
-
-            if has_charges and not hasattr(self, "atomcharges"):
-                self.atomcharges = {}
-
-            if has_spin and not hasattr(self, "atomspins"):
-                self.atomspins = {}
-
-            ones = next(inputfile)
-
-            charges = []
-            spins = []
-            nline = next(inputfile)
-            while not "Sum of" in nline:
-                if has_charges:
-                    charges.append(float(nline.split()[2]))
-
-                if has_spin and has_charges:
-                    spins.append(float(nline.split()[3]))
-
-                if has_spin and not has_charges:
-                    spins.append(float(nline.split()[2]))
-
-                nline = next(inputfile)
-
-            if "Mulliken" in line:
-                if has_charges:
-                    self.atomcharges["mulliken"] = charges
-                if has_spin:
-                    self.atomspins["mulliken"] = spins
+        # Spins may be included in the same section, e.g.
+        #
+        # Mulliken charges and spin densities:
+        #               1          2
+        #     1  O   -0.596188   0.019133
+        #     2  C    0.320624   0.000869
+        #
+        # APT and Lowdin charges are also displayed in this way
+        def extract_charges_spins(prop,header):
+            phrases = [" atomic charges:",
+            " Atomic Charges:",
+            " charges:",
+            " charges with hydrogens summed into heavy atoms:",
+            " atomic charges with hydrogens summed into heavy atoms:",
+            " atomic spin densities:",
+            " charges and spin densities with hydrogens summed into heavy atoms:",
+            " charges and spin densities:"]
+            for phrase in phrases:
+                to_join = [prop,phrase]
+                if "".join(to_join) in line:
+        
+                    has_spin = 'spin' in line or 'Spin' in line
+                    has_charges = 'charges' in line or 'Charges' in line
+
+                    if has_charges and not hasattr(self, "atomcharges"):
+                        self.atomcharges = {}
+
+                    if has_spin and not hasattr(self, "atomspins"):
+                        self.atomspins = {}
+
+                    ones = next(inputfile)
+
+                    charges = []
+                    spins = []
+                    nline = next(inputfile)
 
-            elif "Lowdin" in line:
-                self.atomcharges["lowdin"] = charges
+                    # calculate how many lines need iterating over based on whether property is summed or not
+                    is_sum = 'summed' in line
+                    if is_sum:
+                        n = self.natom - self.nhydrogen
```

Comment:
variable maybe better named `n_atoms`?

Reply:
natom is their variable. I added nhydrogen but copied the naming convention they used for that